File: Project Euler #2/Python/solution2.py
if __name__ == "__main__":
    t = int(input().strip())
    for a0 in range(t):
        n = int(input().strip())
        f = 0
        f1 = 0
        f2 = 1
        sum = 0
        f = f1+f2
        while f < n:
            if f%2 == 0:
                sum += f
            f1 = f2
            f2 = f
            f = f1+f2
        print(sum) 

File: Project Euler #2/CPP/solution2.cpp
#include <iostream>
using namespace std;
int main() {
    int t;
    cin>>t;
    while(t--){
        long long int i, n, f1,f2,f, sum;
        // n <= 4*10^16 , so long long
        cin>>n;
        f1=0;
        f2=1;
        f = f1+f2;
        sum = 0;
        while(f<n){
            if(f%2==0){
                sum += f;
            }
            f = f1+f2;
            f1=f2;
            f2=f;
        }
        cout<<sum<<endl;  
    }
    return 0;
}